### ISSUE TITLE
feature: options profiles

### ISSUE THREAD

**Issue by brainchild0** [flacon/flacon#105]
The user currently can choose a variety of options that affect splitting and transcoding.  Such options include encoding type, encoding options, and CUE and audio file name templates.

In some cases users have multiple targets, for example, laptop, phone, and so on, and need to use different options for each case, especially with respect to encoding type and rate.

Currently, the only way to switch between multiple distinct sets of options is to change each option manually.  Of course this process can be cumbersome and error prone.  It might be helpful if the user could save the current options in a named *profile* that can be persisted independent of other profiles or of the current working set of options.  

---

Possibly related to #57.

**Comment by SokoloffA** [comment]
I was thinking about the profiles. On the one hand, this can be useful to someone, on the other hand, it will complicate the program GUI.  I don't know how popular this feature will be. How many users need it.

But the program can use different configuration files (defined by the command line option). So if you use Linux, you can get it right now using a few shortcuts. In this example we create a profile for the phone. I added instruction to the [FAQ](https://github.com/flacon/flacon/wiki/Frequently-Asked-Questions#in-some-cases-users-have-multiple-targets-for-example-laptop-phone-and-so-on-and-need-to-use-different-options-for-each-case-especially-with-respect-to-encoding-type-and-rate).

**Comment by brainchild0** [comment]
I understand the dilemma.  I can't speak for other users, but use of multiple devices and cloud accounts is increasingly common.


For a new feature, are you mostly concerned about clutter in the graphical interface, or maintainability of the code base?

**Comment by SokoloffA** [comment]
Changes in the code shouldn't cause any problems. 

I am worried about the simplicity and convenience of the graphical interface.
I thought, we could make this feature optional (checkbox on preferences).  By default, it is disabled and the main form looks identical to the current one. But if you enable it, you will see a profiles select. Some like this: 

![Profiles](https://user-images.githubusercontent.com/854935/67633589-bd9c6d00-f8c2-11e9-84b9-5ed66cb41bfd.png)

**Comment by brainchild0** [comment]
I think that a toggle entry for *Show advanced options* in one of the menu bar menus would be a good solution, as would simply a checkbox on the side panel, or a roll-down panel if the widget kit has support.

**Comment by brainchild0** [comment]
Hiding advanced options is a compelling concern if they are numerous or complicated.  Such might become the case if you further develop this feature or others.  In the picture you propose, the added bloat to the interface is very minimal, with the existing features consuming most of the space.

**Comment by brainchild0** [comment]
Another option would be to put a checkbox immediately above the new drop down menu. The checkbox could be unchecked by default and labeled "Use profiles", and the menu could be disabled when the checkbox is unchecked.  This combination of interactions would make the extra features less distracting to the user not in need of them.

**Comment by SokoloffA** [comment]
Hi, guys! Good news! 
I added support for profiles. Now I'm waiting for the translators to finish the translations, and in a couple of weeks I'll make a release.

**Comment by brainchild0** [comment]
I am now running version 6.0.0.  I have not found the new feature for profiles.  Where in the interface should I look for it?

**Comment by SokoloffA** [comment]
>I found the add and remove buttons in the preferences box, but I was confused because the word "format" is still in use (in the English version), though it now apparently means profile.

I did it on purpose. That's my logic:

* The program is many years old and it had a only formats list. 
* Requests to add profiles are not frequent, most users are satisfied with a simple list of formats.
* Most users use programs with default settings. By default, the program has one profile per format. So new profile list looks like previous format list.

In order not to confuse old users, I decided to have left the look and UX of the program as similar as possible to previous versions. If you don't add your own profiles, the program looks and works like before.

Maybe I'm wrong, I'm open to any ideas about it.

P.S.
_If you don't mind, I suggest moving the discussion to the program's issue tracker. Perhaps someone else will join the discussion. I'll copy a couple of messages to a new issue._
